### PR TITLE
synapse: 1.34.0 -> 1.35.0

### PR DESCRIPTION
**Pull request by sumnerevans**
###### Motivation for this change

Tracking PR for Synapse 1.35.0.

https://github.com/matrix-org/synapse/blob/develop/CHANGES.md#synapse-1350rc1-2021-05-25
https://github.com/matrix-org/synapse/blob/v1.35.0rc2/CHANGES.md#synapse-1350rc2-2021-05-27
https://github.com/matrix-org/synapse/releases/tag/v1.35.0rc3
https://github.com/matrix-org/synapse/releases/tag/v1.35.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/matrix-synapse/default.nix [hunk 1]
```diff
@@ -66,7 +67,11 @@ buildPythonApplication rec {
   doCheck = !stdenv.isDarwin;
 
   checkPhase = ''
-    ${lib.optionalString (!enableRedis) "rm -r tests/replication # these tests need the optional dependency 'hiredis'"}
+    ${lib.optionalString (!enableRedis) ''
+      # these tests need the optional dependency 'hiredis'
+      rm -r tests/replication
+      rm -r tests/module_api
```
Comment (sumnerevans) [inline pkgs/servers/matrix-synapse/default.nix:73]: Note I had to remove some more tests that have dependency on redis.

Comment (andir) [reply pkgs/servers/matrix-synapse/default.nix:73]: They seem rather core to the UX of synapse: https://github.com/matrix-org/synapse/blob/develop/tests/module_api/test_api.py#L52-L80

Isn't there a way to skip those tests that require redis?

**Comment by andir** [comment]
Might be worth to create a follow-up PR that updates our systemd units: https://github.com/matrix-org/synapse/pull/9803/files

File: pkgs/servers/matrix-synapse/default.nix (hunk 1 above)
Comment (sumnerevans) [reply pkgs/servers/matrix-synapse/default.nix]: Removing both of these exceptions seems to have actually worked. I'm not sure why it failed for me 🤷.

**Comment by sumnerevans** [comment]
> Might be worth to create a follow-up PR that updates our systemd units: https://github.com/matrix-org/synapse/pull/9803/files

I'd like to get #120260 in before making those updates as there will be a lot of merge conflicts otherwise.

**Review by andir — APPROVED** [review]

**Comment by Ma27** [comment]
FYI back when I updated matrix-synapse, I had a "backport-by-default" policy. I'll try to briefly explain the why: synapse regularly has CVEs that have to be patched and those are usually released on new minors. From my experience it's less painful to have regular updates rather than a "big" update when it's necessary.

Also, there were non-trivial infra changes in the past that made synapse upgrades mandatory under certain circumstances, see for instance https://github.com/matrix-org/synapse/blob/develop/UPGRADE.rst#3pid-verification-changes.

cc @domenkozar for the `The process '/usr/bin/git' failed with exit code 1`-error. Not sure what's that about :)

**Comment by sumnerevans** [comment]
@Ma27 thanks for explaining. I'll add that to future PRs for `matrix-synapse` upgrades.

Also, I've been updating matrix-synapse the the bleeding edge for my own servers, so that's why I figured I should just do it in nixpkgs as well. If you would rather handle it yourself, let me know and I'll step back :)

**Pull request merged**